src/morphosamplers/utils.py

```diff
@@ -0,0 +1,36 @@
+from typing import Optional
+
+import numpy as np
+
+
+def calculate_y_vectors_from_z_vectors(
+    z: np.ndarray,
+    initial_y_vector: Optional[np.ndarray] = None
+) -> np.ndarray:
+    """Calculate y vectors starting from z vectors.
+
+    This function will return the set of unit vectors perpendicular to the z-vectors which are
+    maximally coaxial to their neighbours. It assumes that z vectors (n, 3) vary
+    smoothly with increasing n.
```

Comment:
I _think_ this is correct?

```suggestion
    smoothly and monotonically with increasing n.
```

Reply:
No, you can have a "snakey" filament and it will work just fine. What won't "work" is if you have z vector `i` and z vector `i + 1` be basically unrelated, then the results will be just as unrelated... which I guess we could even leave out :P

Reply:
agree monotonically isn't quite right here for the z vectors, it is right for n but seems unnecessary